TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java

```diff
@@ -170,24 +155,33 @@ private static IConnectionCaptureFactory getConnectionCaptureFactory(Parameters
         // Resist the urge for now though until it comes in as a request/need.
         if (params.traceDirectory != null) {
             return new FileConnectionCaptureFactory(nodeId, params.traceDirectory, params.maximumTrafficStreamSize);
-        } else if (params.kafkaPropertiesFile != null) {
-            if (params.kafkaConnection != null || params.mskAuthEnabled) {
-                log.warn("--kafkaConnection and --enableMSKAuth options are ignored when providing a Kafka properties file (--kafkaConfigFile) ");
-            }
-            return getKafkaConnectionFactory(nodeId, params.kafkaPropertiesFile, params.maximumTrafficStreamSize);
         } else if (params.kafkaConnection != null) {
             var kafkaProps = new Properties();
+            kafkaProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+            kafkaProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer");
+            // Property details: https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#delivery-timeout-ms
+            kafkaProps.put(ProducerConfig.ACKS_CONFIG, "1");
+            kafkaProps.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 1000);
+            kafkaProps.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, 500);
+            kafkaProps.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, 3000);
+
+            if (params.kafkaPropertiesFile != null) {
+                try {
+                    kafkaProps.load(new FileReader(params.kafkaPropertiesFile));
+                } catch (IOException e) {
+                    log.error("Unable to locate provided Kafka producer properties file path: " + params.kafkaPropertiesFile);
+                    throw e;
+                }
+            }
```

Comment:
Can you write a test to show/document how precedence will work for these?  I would have expected to load the file BEFORE loading the other properties so that the more granular options could override the file rather than the other way around.